### ISSUE TITLE
Search does not cover all incomplete search terms

### ISSUE THREAD

**Issue by rmarcalo** [ScottLogic/StockFlux#671]
During my user testings I've noticed that the search does not always find stocks by using incomplete search terms. Here is an example below:
![search](https://cloud.githubusercontent.com/assets/13066377/13842332/763b32f0-ec25-11e5-9371-a93e4c269407.png)

But if I type Tesco the correct stock appears. This always happens for "Tesc" , or something like "Vodaf" (for vodafone), which makes me believe that after a certain number of characters (I believe it's 4) the search simply waits for the entire keyword to be typed-in, which might have been an intentional implementation decision. I believe, however, that the ideal interaction would be that the search looks for incomplete search terms at all times, as it has confused some of my tested users into thinking the stock does not exist.

**Comment by owennw** [comment]
Yes we've seen this too. It seems that Quandl doesn't use [fuzzy search](https://en.wikipedia.org/wiki/Approximate_string_matching) and we're just using Quandl's search API.

There is an issue to download all of the stock symbols for offline usage (#559) so that could be the time to introduce a better search manually.
